devicetypes/fibargroup/fibaro-walli-dimmer-switch.src/fibaro-walli-dimmer-switch.groovy

```diff
@@ -0,0 +1,525 @@
+}
+
+def off() {
+	encapSequence([
+			zwave.basicV1.basicSet(value: 0x00),
+			zwave.switchMultilevelV1.switchMultilevelGet(),
+	], 1000)
+}
+
+def setLevel(level, rate = null) {
+	if(level > 99) level = 99
+	encapSequence([
+			zwave.basicV1.basicSet(value: level),
+			zwave.switchMultilevelV1.switchMultilevelGet()
+	], 1000)
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	log.debug "ping() called"
+	refresh()
+}
+
+def refresh() {
+	log.debug "refresh()"
+
+	encapSequence([
+			zwave.switchMultilevelV1.switchMultilevelGet(),
+			meterGet(scale: 0),
+			meterGet(scale: 2),
+	], 1000)
+}
+
+def reset() {
+	encapSequence([
+			meterReset(),
+			meterGet(scale: 0)
+	])
+}
+
+def meterGet(scale) {
+	zwave.meterV2.meterGet(scale)
+}
+
+def meterReset() {
+	zwave.meterV2.meterReset()
+}
+
+private encapSequence(cmds, Integer delay=250) {
+	delayBetween(cmds.collect{ encap(it) }, delay)
+}
+
+private encap(physicalgraph.zwave.Command cmd) {
+	if (zwaveInfo?.zw?.contains("s")) {
+		secEncap(cmd)
+	} else {
+		log.debug "no encapsulation supported for command: $cmd"
+		cmd.format()
+	}
+}
+
+private secEncap(physicalgraph.zwave.Command cmd) {
+	log.debug "encapsulating command using Secure Encapsulation, command: $cmd"
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private getParameterMap() {[
+	[
+		name: "LED frame – colour when ON", key: "ledFrame–ColourWhenOn", type: "enum",
+		parameterNumber: 11, size: 1, defaultValue: 1,
+		values: [
+			0: "LED disabled",
+			1: "White",
+			2: "Red",
+			3: "Green",
+			4: "Blue",
+			5: "Yellow",
+			6: "Cyan",
+			7: "Magenta",
+			8: "colour changes smoothly depending on the measured power",
+			9: "colour changes in steps depending on the measured power"
+		],
+		description: "LED colour when the device is ON. When set to 8 or 9, LED frame colour will change depending on the measured power and parameter 10. Other colours are set permanently and do not depend on the power consumption."
+	],
+	[
+		name: "LED frame – colour when OFF", key: "ledFrame–ColourWhenOff", type: "enum",
+		parameterNumber: 12, size: 1, defaultValue: 0,
+		values: [
+			0: "LED disabled",
+			1: "White",
+			2: "Red",
+			3: "Green",
+			4: "Blue",
+			5: "Yellow",
+			6: "Cyan",
+			7: "Magenta"
+		],
+		description: "LED colour when the device is OFF."
+	],
+	[
+		name: "LED frame – brightness", key: "ledFrame–Brightness", type: "range",
+		parameterNumber: 13, size: 1, defaultValue: 100,
+		range: "0..102",
+		description: "Adjust the LED frame brightness. " +
+				"101 - brightness directly proportional to set level, " +
+				"102 - brightness inversely proportional to set level"
+	],
+	[
+		name: "Manual control – dimming step size", key: "manualControl–DimmingStepSize", type: "range",
+		parameterNumber: 156, size: 1, defaultValue: 1,
+		range: "1..99",
+		description: "Percentage value of the dimming step during the manual control. " +
```

Comment:
How come some of these lists have manual newlines but some don't?

Also this wording is a bit confusing to me. Can you explain what the 1-4 values actually do?

Reply:
@greens 
"new lines" - are for the sake of readability.
"1-4 values" - thanks for pointing that out. I verified it on zwave alliance:
There are two certification documents for that device (FGWDEU-11) and in a first one they don't occur for parameter number 156, they are only related to "Alarm configuration" parameters. They will be removed here.

Reply:
ok, fixed